### PR TITLE
Save --privileged state

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@mheon @baude @umohnani8 @TomSweeneyRedHat PTAL
I had this patch as part of the --no-new-privs, but really no reason to wait.  Fixes storing of the --privileged flag, and adds some missing tests.

**Comment by mheon** [comment]
LGTM

File: test/podman_run_security.bats [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+function setup() {
+    copy_images
+}
+
+@test "run privileged test" {
+    cap=$(grep CapEff /proc/self/status | cut -f2 -d":")
+
+    run ${PODMAN_BINARY} ${PODMAN_OPTIONS} run --privileged ${ALPINE}  grep CapEff /proc/self/status
```
Comment (umohnani8) [inline test/podman_run_security.bats]: Any reason why no doing bash -c here?

**Comment by umohnani8** [comment]
LGTM, just 1 clarification question.

File: libpod/options.go [hunk 2]
```diff
@@ -268,6 +268,18 @@ func WithShmSize(size int64) CtrCreateOption {
 	}
 }
 
+// WithPrivileged sets the privileged flag in the container runtime
+func WithPrivileged(privileged bool) CtrCreateOption {
+	return func(ctr *Container) error {
+		if ctr.valid {
```
Comment (TomSweeneyRedHat) [inline libpod/options.go]: This just looks odd, we've a valid container, so we raise an error.  Could we change this field name to finalized vs valid?  I don't know if that works or not, just thinking aloud.  Not a needed adjustment for this PR.

File: test/podman_run_security.bats (hunk 1 above)
Comment (rhatdan) [reply test/podman_run_security.bats]: I am actually running the grep command in the container, as opposed to piping the output to the grep command.  Bottom line, no reason to do it.

**Comment by rhatdan** [comment]
@rh-atomic-bot r=mheon

File: test/podman_run_security.bats (hunk 1 above)
Comment (umohnani8) [reply test/podman_run_security.bats]: Then aren't you doing the same thing for the 2 tests below? No point in have bash -c there as well.

**Comment by rhatdan** [comment]
@rh-atomic-bot retry

**Comment by rhatdan** [comment]
Tests finally pass.

@rh-atomic-bot r=mheon

**Pull request closed without merging**